core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -151,36 +148,19 @@ object ClientMiddleware {
             trace = res.trace
             traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty)).mapK(trace)
             newReq = req.withHeaders(traceHeaders ++ req.headers)
-            resp <- poll(client.run(newReq)).guaranteeCase { outcome =>
-              (outcome match {
-                case Outcome.Succeeded(fa) =>
-                  fa.flatMap { resp =>
-                    Resource
-                      .eval {
-                        span.addAttributes(
-                          response(resp, allowedResponseHeaders) ++
-                            additionalResponseAttributes(resp.responsePrelude)
-                        )
-                      }
-                      .mapK(trace)
-                  }
-                case Outcome.Errored(e) =>
-                  Resource.eval(span.recordException(e)).mapK(trace)
-                case Outcome.Canceled() =>
-                  // Canceled isn't always an error, but it generally is for http
-                  // TODO: decide if this should add "error", we do for the server side.
-                  Resource.eval(span.addAttribute(CustomAttributes.Canceled(true))).mapK(trace)
-              }) >> Resource.eval(span.addAttribute(CustomAttributes.exitCase(outcome))).mapK(trace)
+
+            _ <- Resource.onFinalizeCase { ec =>
+              span.addAttribute(CustomAttributes.exitCase(ec.toOutcome))
```

Comment:
`Span` has a [status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status) in the OpenTelemetry.  So `exit.case` looks like a duplication to me. 
Perhaps we can drop this explicit attribute?

Reply:
let's go for it. there seems to be a lot of vestigial stuff from the migration (which is my fault 😅)

Reply:
do you want to update this PR to do that here, or do it in a separate PR?

Reply:
I will make a follow up PR.